lib/IDE/CompletionInstance.cpp

```diff
@@ -320,6 +320,11 @@ bool CompletionInstance::performNewOperation(
   if (DiagC)
     CI.addDiagnosticConsumer(DiagC);
 
+  SWIFT_DEFER {
+    if (DiagC)
+      CI.removeDiagnosticConsumer(DiagC);
```

Comment:
`CI` is a reference to `TheInstance` that can be `std::move`d before leaving this scope.

Reply:
@rintaro Ah, good catch!

Reply:
I'm nervous about having a lifetime-ending operation written in the middle of a scope.

Reply:
I've gone ahead and moved the call to `removeDiagnosticConsumer` back to where it was originally, as the early exit cases should be handled by the destruction of the `CompilerInstance` anyway.